packages/ra-core/src/controller/list/useListController.ts

```diff
@@ -168,6 +169,37 @@ export const useListController = <RecordType extends RaRecord = any>(
         name: getResourceLabel(resource, 2),
     });
 
+    const { data: allData } = useGetList<RecordType>(
```

Comment:
You'd better use the data provider directly in the callback, or using react-query's useMutation. useGetList is really designed for queries triggered on mount.

Reply:
I agree the concept but I can't. I we use the `getList` function directly in the callback, we can't have the full data array in the context. If we don't have this array, we can't know if we must display the "SELECT ALL" button.